compiler/rustc_infer/src/infer/error_reporting/nice_region_error/static_impl_trait.rs

```diff
@@ -356,20 +356,87 @@ pub fn suggest_new_region_bound(
+                                            finish_bracket(
+                                                &mut spans_suggs,
+                                                name,
+                                                &mut bracket_span,
+                                                &mut brackets,
+                                            );
+                                            spans_suggs.push((p.span, name.to_string()));
+                                        }
+                                        MissingLifetimeKind::Ampersand => {
+                                            finish_bracket(
+                                                &mut spans_suggs,
+                                                name,
+                                                &mut bracket_span,
+                                                &mut brackets,
+                                            );
+                                            spans_suggs
+                                                .push((p.span.shrink_to_hi(), format!("{name} ")));
+                                        }
+                                        MissingLifetimeKind::Comma => {
+                                            finish_bracket(
+                                                &mut spans_suggs,
+                                                name,
+                                                &mut bracket_span,
+                                                &mut brackets,
+                                            );
+                                            spans_suggs
+                                                .push((p.span.shrink_to_hi(), format!("{name}, ")));
+                                        }
+                                        MissingLifetimeKind::Brackets => {
+                                            if bracket_span.is_some_and(|span| span != p.span) {
+                                                finish_bracket(
+                                                    &mut spans_suggs,
+                                                    name,
+                                                    &mut bracket_span,
+                                                    &mut brackets,
+                                                );
+                                            }
+                                            bracket_span = Some(p.span);
+                                            brackets += 1;
+                                        }
+                                    }
+                                }
+                            }
+                            finish_bracket(
+                                &mut spans_suggs,
+                                name,
+                                &mut bracket_span,
+                                &mut brackets,
+                            );
+                            spans_suggs
+                        }
```

Comment:
I think it's better to define a new function for these lines since new implementation has many lines. Could you address this?

Reply:
Yes, it looked so ugly to me 😅